### PR TITLE
GitHub issue guide pr

### PR DESCRIPTION
**Pull request by smellycloud**
I realize that I haven't used my own guidelines while naming this branch. Nice. This is only a draft template. I'm open to thoughts and criticisms. Will add examples of poorly written issues after I'm done searching.

### CODE REVIEW

File: documentation/github_feature_request_template.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+## **Labels**
+Assign appropriate labels to categorize the feature request (e.g., enhancement, new feature).
+
+## **Assignees**
+Assign the feature request to relevant team members.
+
+## **Milestones**
+- [ ] Task 1
+- [ ] Task 2
+- [ ] Task 3
```
Comment (Polichinel) [inline documentation/github_feature_request_template.md:36]: Yes, but there's a difference between asking people to add a todo in the feature request (which we should certainly encourage) and connecting a feature request (issue) to a milestone, right (which we should also encourage). But this looks like a bit of both, no? Or am I missing something?

File: documentation/github_feature_request_template.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+## **Linked Issues**
+Link to any related issues or pull requests.
+
+## **Error Handling & Logging**
+- Ensure that any potential errors related to the feature are considered and documented.
+- Include logging requirements if applicable.
+
+## **README Generation**
+- Ensure that the features are documented in the project's relevant README file.
+
```
Comment (Polichinel) [inline documentation/github_feature_request_template.md:47]: Also something like this:

## **ADR  Generation**
- Ensure that all decisions that manifest as part of the feature implementations get appropriately documented as one or more ADRs

File: documentation/github_bugfix_issue_template.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+## **Linked Issues**
+Link to any related issues or pull requests.
+
+## **Error Handling & Logging**
+- Ensure that any error messages are clear and actionable.
+- Include logs if applicable.
+
+## **Future Flexibility**
+- Consider how this issue might affect future development and document any potential impacts.
+
```
Comment (Polichinel) [inline documentation/github_bugfix_issue_template.md:51]: Also something like this:

## **ADR  Update/Generation**
- Consider whether solving this issue should lead to the updating and/or generation of one or more ADRs to ensure the ADRs still correctly represent the actual codebase.

File: documentation/guide_to_issues_on_github.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
+# Guide to GitHub Issues
+
+## Description
+To ensure that our team handles GitHub issues correctly and consistently. This guide will serve as a quick reference for all team members, new and experienced, by outlining best practices for creating, addressing, and resolving issues. Our intention is to streamline our workflow, reduce misunderstandings, and ensure that all issues are addressed swiftly and without the need for follow-up questions. If you are new to GitHub and issues, start [here](https://docs.github.com/en/issues).
```
Comment (Polichinel) [inline documentation/guide_to_issues_on_github.md:4]: Really neat and precise formulation

File: documentation/guide_to_issues_on_github.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    * Actual Behavior: What actually happens.
+    * Screenshots/Logs: Attach relevant screenshots or logs.
+
+3. **Guidelines for Labeling Issues**
+    * Use predefined labels such as `bug`, `documentation`, `help wanted`, `improvement`. See [label guide](https://github.com/prio-data/views_pipeline/labels) for more information.
+    * Ensure labels are applied consistently.
+
+4. **Assigning Issues**
+    * Assign issues to the appropriate team members based on expertise.
+    * Use the Assignees feature in GitHub.
```
Comment (Polichinel) [inline documentation/guide_to_issues_on_github.md:35]: I think we can add that usually 2 people should be assigned - and for very substantial changes to working solutions, more might be needed.

File: documentation/guide_to_issues_on_github.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    * Summarize the solution and any important details.
+
+4. **Reopening Issues**
+    * Reopen if the problem persists.
+    * Provide additional context or information.
+
+### Common Pitfalls
+
+1. **Overloading Issues**
+   * Avoid combining multiple problems or features into a single issue. This can make it difficult to track and resolve each aspect effectively.
```
Comment (Polichinel) [inline documentation/guide_to_issues_on_github.md:78]: Very important

File: documentation/guide_to_issues_on_github.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+2. **Ignoring Conversations**
+   * GitHub Issues are not just for reporting problems; they are also a forum for discussion. Engaging in conversations, providing feedback, and asking questions can lead to better solutions.
+
+3. **Not Using Milestones**
+   * Milestones help track progress and set expectations for when issues should be resolved. Neglecting to use them can lead to a lack of direction and missed deadlines.
```
Comment (Polichinel) [inline documentation/guide_to_issues_on_github.md:84]: We don't do this enough yet - but you're right

**Review by Polichinel — CHANGES_REQUESTED** [review]
I really like the division of docs in this draft, and I find the structure and language to be very clear and precise.

I've added a few minor comments

File: documentation/github_feature_request_template.md (hunk 1 above)
Comment (smellycloud) [reply documentation/github_feature_request_template.md]: My bad. I actually meant actionable tasks.

File: documentation/github_feature_request_template.md (hunk 2 above)
Comment (smellycloud) [reply documentation/github_feature_request_template.md:47]: Added

File: documentation/guide_to_issues_on_github.md (hunk 5 above)
Comment (smellycloud) [reply documentation/guide_to_issues_on_github.md]: Done

File: documentation/github_bugfix_issue_template.md (hunk 3 above)
Comment (Polichinel) [reply documentation/github_bugfix_issue_template.md:52]: nice

File: documentation/github_feature_request_template.md (hunk 2 above)
Comment (Polichinel) [reply documentation/github_feature_request_template.md:47]: nice

**Review by Polichinel — APPROVED** [review]
Looks good

**Pull request merged**